### ISSUE TITLE
CI: upgrade to Ubuntu 22.04

### ISSUE THREAD

**Issue by cfallin** [bytecodealliance/wasmtime#5408]
In #5407 we pinned to 20.04 because of issues such as missing `libclang1-9` in [this CI run](https://github.com/bytecodealliance/wasmtime/actions/runs/3659854687/jobs/6186416352) with an Ubuntu 22.04 image.

We should resolve all issues that appear on 22.04 and upgrade to the new version when possible.

**Comment by alexcrichton** [comment]
After https://github.com/bytecodealliance/wasmtime/pull/5410 the only 20.04 builder is the wasi-nn builder, and @abrown [this is a failure on the 22.04 builder](https://github.com/bytecodealliance/wasmtime/actions/runs/3659854687/jobs/6186417666) which gives:

```
Warning: apt-key is deprecated. Manage keyring files in trusted.gpg.d instead (see apt-key(8)).
OK
deb https://apt.repos.intel.com/openvino/2022 jammy main

WARNING: apt does not have a stable CLI interface. Use with caution in scripts.

Get:1 https://packages.microsoft.com/ubuntu/22.04/prod jammy InRelease [10.5 kB]
Hit:2 http://azure.archive.ubuntu.com/ubuntu jammy InRelease
Get:3 http://azure.archive.ubuntu.com/ubuntu jammy-updates InRelease [114 kB]
Get:4 http://azure.archive.ubuntu.com/ubuntu jammy-backports InRelease [99.8 kB]
Get:5 http://azure.archive.ubuntu.com/ubuntu jammy-security InRelease [110 kB]
Get:6 https://packages.microsoft.com/ubuntu/22.04/prod jammy/main arm64 Packages [16.7 kB]
Get:7 https://packages.microsoft.com/ubuntu/22.04/prod jammy/main armhf Packages [10.6 kB]
Get:8 https://packages.microsoft.com/ubuntu/22.04/prod jammy/main amd64 Packages [60.1 kB]
Get:9 http://azure.archive.ubuntu.com/ubuntu jammy-updates/main amd64 Packages [758 kB]
Get:10 http://azure.archive.ubuntu.com/ubuntu jammy-updates/main Translation-en [169 kB]
Hit:11 https://ppa.launchpadcontent.net/ubuntu-toolchain-r/test/ubuntu jammy InRelease
Get:12 http://azure.archive.ubuntu.com/ubuntu jammy-updates/main amd64 c-n-f Metadata [11.4 kB]
Get:13 http://azure.archive.ubuntu.com/ubuntu jammy-updates/restricted amd64 Packages [488 kB]
Get:14 http://azure.archive.ubuntu.com/ubuntu jammy-updates/restricted Translation-en [74.9 kB]
Get:15 http://azure.archive.ubuntu.com/ubuntu jammy-updates/universe amd64 Packages [761 kB]
Get:16 http://azure.archive.ubuntu.com/ubuntu jammy-updates/universe Translation-en [128 kB]
Get:17 http://azure.archive.ubuntu.com/ubuntu jammy-updates/universe amd64 c-n-f Metadata [14.0 kB]
Get:18 http://azure.archive.ubuntu.com/ubuntu jammy-updates/multiverse amd64 Packages [7300 B]
Get:19 http://azure.archive.ubuntu.com/ubuntu jammy-backports/main amd64 Packages [3324 B]
Get:20 http://azure.archive.ubuntu.com/ubuntu jammy-backports/main Translation-en [1580 B]
Get:21 http://azure.archive.ubuntu.com/ubuntu jammy-backports/main amd64 c-n-f Metadata [272 B]
Get:22 http://azure.archive.ubuntu.com/ubuntu jammy-backports/universe amd64 Packages [6740 B]
Get:23 http://azure.archive.ubuntu.com/ubuntu jammy-backports/universe amd64 c-n-f Metadata [348 B]
Get:24 http://azure.archive.ubuntu.com/ubuntu jammy-security/main amd64 Packages [525 kB]
Get:25 http://azure.archive.ubuntu.com/ubuntu jammy-security/main Translation-en [113 kB]
Get:26 http://azure.archive.ubuntu.com/ubuntu jammy-security/main amd64 c-n-f Metadata [7352 B]
Get:27 http://azure.archive.ubuntu.com/ubuntu jammy-security/restricted amd64 Packages [452 kB]
Get:28 http://azure.archive.ubuntu.com/ubuntu jammy-security/restricted Translation-en [69.2 kB]
Get:29 http://azure.archive.ubuntu.com/ubuntu jammy-security/universe amd64 Packages [622 kB]
Get:30 http://azure.archive.ubuntu.com/ubuntu jammy-security/universe Translation-en [82.7 kB]
Get:31 http://azure.archive.ubuntu.com/ubuntu jammy-security/universe amd64 c-n-f Metadata [11.0 kB]
Ign:32 https://apt.repos.intel.com/openvino/2022 jammy InRelease
Err:33 https://apt.repos.intel.com/openvino/2022 jammy Release
  404  Not Found [IP: 184.50.211.30 443]
Reading package lists...
E: The repository 'https://apt.repos.intel.com/openvino/2022 jammy Release' does not have a Release file.
Error: Process completed with exit code 100.
```

Upgrading to 22.04 is not of the utmost urgency at all, but might be good to forward to folks at Intel I think hosting openvino there?

**Comment by abrown** [comment]
I saw this in a different repository as well (failure to pull OpenVINO packages for Ubuntu 22.04) and I contacted the packaging folks for OpenVINO. They said the next release of OpenVINO (2022.3, I believe, so that should be coming soon!) will have the Ubuntu 22.04 packages. Can we wait for a few weeks to unpin the CI task? Feel free to assign this to me if the wasi-nn tests are the only task preventing us from moving to 22.04 and I will fix this once I see the new packages.
